doc/doxygen/src/advanced-build-system-tricks.md

```diff
@@ -40,6 +40,78 @@ You can configure your own files that will be parsed by the build system main
 * Override default targets
 
 
+Handling multiple boards with udev-rules                {#multiple-boards-udev}
+========================================
+
+When developing and working with multiple boards the default `PORT` configuration
+for a particular board might not apply anymore so `PORT` will need to be specified
+whenever calling `make term/test`. This can also happen if multiple `DEBUGGERS/PROGRAMMERS`
+are present so `DEBUG_ADAPTER_ID` will also need to be passed. Keeping track of
+this will become annoying.
+
+We can use custom `udev` rules to define `SYMLINKS` between the boards
+serial port (`dev/tty-board-name`) and the actual serial port (dev/ttyACM* or
+other). With this we can query the rest of the boards serial `dev` information
+(`DEBUG_ADAPTER_ID`, `PORT`, etc.) to always flash and open a terminal on the
+correct port.
+
+Procedure:
+
+- use `udevadm info /dev/ttyACM0` to query the udev database for information on 
+  device on port `/deb/ttyACM0`.
+
+  or use `udevadm info --attribute-walk --name /dev/ttyACM0` for more detailed
+  output when first level of information isn't enough
+
+- create a udev rule with information of the device and one parent to create a
+  matching rule in `/etc/udev/rules.d/70-riotboards.rules`.
+
+~~~~~~~~~~~~~~~~~~~
+    # samr21-xpro
+    SUBSYSTEM=="tty", SUBSYSTEMS=="usb", ATTRS{idVendor}=="03eb", \
+    ATTRS{idProduct}=="2111", ATTRS{manufacturer}=="Atmel Corp.", \
+    ATTRS{serial}=="ATML2127031800004957", SYMLINK+="riot/tty-samr21-xpro"
+~~~~~~~~~~~~~~~~~~~
+
+- reload rules: `udevadm control --reload-rules`
+
+- Boards `PORT` are symlinked to /dev/riot/tty-`board-name`.
+
+- Create a `makefile.pre` that will query the real `PORT` and the `DEBUG_ADAPTER_ID`
+  from the `SYMLINK` info
+
+~~~~~~~~~~~~~~~~~~~
+    PORT = /dev/riot/tty-$(BOARD)
+    DEBUG_ADAPTER_ID = $(\
+        shell udevadm info -q property $(PORT) |\
+        sed -n ’/ID_SERIAL_SHORT/ {s/ID_SERIAL_SHORT=//p}’)
+~~~~~~~~~~~~~~~~~~~
+
+- You can now add `makefile.pre` to `RIOT_MAKEFILES_GLOBAL_PRE` as an environment
```

Comment:
If set globally, I would recommend documenting to put  a variable to enable it as not all boards may have the udev rule configured.